usr/share/rear/output/default/94_grub2_rescue.sh

```diff
@@ -110,6 +110,9 @@ else
     grub-mkconfig -o $grub_conf
 fi
 
+#Finding UUID of filesystem containing /boot 
+grub_boot_uuid=$(df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value)
```

Comment:
@gdha When `/boot` is not mounted, `df` will return `/` as a filesystem. UUID returned will be root UUID instead of `/boot` (which is not good).
`grub-mkconfig` has the same behavior !!! it uses `grub2-probe --target=device /boot` to get the boot device (`/dev/sda1`). If `/boot` is unmounted, this command return the root device instead.
So `/boot` should always be mounted to been able to use `grub-mkconfig` properly (automatically used after a kernel upgrade for example).

Anyway, your point is correct, we could check if a directory `/boot/grub*` exists. If yes, this mean that `/boot` is mounted or `/boot `is part of `/ `FS.
```
if ! ls -ld /boot/grub* > /dev/null 2>&1 ; then  
    echo "Error: /boot is empty, or may be not mounted..."
    exit
else 
    grub_boot_uuid=$(df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value)
fi
```
Your feedback is welcomed